src/parsers.rs

```diff
@@ -15,41 +15,38 @@ use nom::{
     bytes::complete::{tag, take_while, take_while_m_n},
     character::complete::one_of,
     character::is_digit,
-    combinator::{map, opt},
-    sequence::preceded,
+    combinator::{map, not, opt},
+    sequence::{preceded, terminated},
     IResult,
 };
 
-use crate::{Date, DateTime, Time};
+use crate::{Date, DateTime, Duration, Time};
 
 #[cfg(test)]
 mod tests;
 
+// UTILITY
+
 fn take_n_digits(i: &[u8], n: usize) -> IResult<&[u8], u32> {
     let (i, digits) = take_while_m_n(n, n, is_digit)(i)?;
 
     let s = str::from_utf8(digits).expect("Invalid data, expected UTF-8 string");
     let res = s
         .parse()
-        .expect("Invalid string, expected ASCII reprensation of a number");
+        .expect("Invalid string, expected ASCII reprensentation of a number");
```

Comment:
Good one :D

Reply:
How did I manage that?! 🙈 

Fixed.

Reply:
Nooo, _I_ managed to mistype that. you fixed it!

Reply:
I still left an extra n in that string even though I wrote it correctly in another function! :smile:

Double fixed it now though!